stingray/pulse/accelsearch.py

```diff
@@ -0,0 +1,283 @@
+            print(*powers_to_search, file=fobj)
+
+        candidate = powers_to_search > detlev
+        rs = rf[candidate]
+        cand_powers= powers_to_search[candidate]
+        for i in range(len(rs)):
+            r = rs[i]
+            cand_power = cand_powers[i]
+            candidate_powers.append(cand_power)
+            candidate_rs.append(r)
+            candidate_js.append(j)
+    if debug:
+        fobj.close()
+    return candidate_rs[1:], candidate_js[1:], candidate_powers[1:]
+
+
+def accelsearch(times, signal, delta_z=1, fmin=1, fmax=1e32,
+                GTI=None, zmax=100, candidate_file=None, ref_time=0,
+                debug=False, interbin=False):
+    """Find pulsars with accelerated search.
+
+    The theory behind these methods is described in Ransom+02, AJ 124, 1788.
+
+    Parameters
+    ----------
+    times : array of floats
+        An evenly spaced list of times
+    signal : array of floats
+        The light curve, in counts; same length as ``times``
+    Other parameters
+    ----------------
+    delta_z : float
+        The spacing in ``z`` space
+    fmin : float, default 1.
+        Minimum frequency to search
+    fmax : float, default 1e32
+        Maximum frequency to search
+    GTI : ``[[gti00, gti01], [gti10, gti11], ...]``, default None
+        Good Time Intervals. If None, it assumes the full range
+        ``[[time[0] - dt / 2 -- time[-1] + dt / 2]]``
+    zmax : int, default 100
+        Maximum frequency derivative to search (pos and neg), in bins.
+        It corresponds to ``fdot_max = zmax / T**2``, where ``T`` is the
+        length of the observation.
+    candidate_file : str, default None
+        Save the final candidate table in this file.
+    ref_time : float, default 0
+        Reference time for the times
+    """
+    times = np.asarray(times)
+    signal = np.asarray(signal)
+    dt = times[1] - times[0]
+    if GTI is not None:
+        GTI = np.asarray(GTI)
+        # Fill in the data with a constant outside GTI
+        gti_mask = create_gti_mask(times, GTI)
+        bti_mask = ~gti_mask
+        signal[bti_mask] = np.median(signal[gti_mask])
+    else:
+        GTI = np.array(
+            [[times[0] - dt /2, times[-1] + dt / 2]])
+
+    n_photons = np.sum(signal)
+    spectr = np.fft.fft(signal)
+    freq = np.fft.fftfreq(len(spectr), dt)
+    T = times[-1] - times[0] + dt
+
+    freq_intv_to_search = (freq >= fmin) & (freq < fmax)
+    log.info("Starting search over full plane...")
+    start_z = -zmax
+    end_z = zmax
+    range_z = np.arange(start_z,end_z, delta_z)
+    log.info("min and max possible r_dot: {}--{}".format(delta_z/T**2,
+                                                         np.max(range_z)/T**2))
+    freqs_to_search = freq[freq_intv_to_search]
+
+    candidate_table = Table(
+        names=['time', 'power', 'prob', 'frequency', 'fdot', 'fddot'],
+        dtype=[float] * 6)
+
+    detlev = detection_level(freqs_to_search.size, epsilon=0.015)
```

Comment:
so does that mean you're setting the detection level to 15%? That seems maybe rather high?

Reply:
It's 1.5%! I set it high because I want to see if, e.g., there is a marginal detection in multiple intervals at the same frequency